src/program.rs

```diff
@@ -207,6 +217,132 @@ where
     };
 }
 
+/// Create a markdown file with documentation for the specified FileId.
+fn output_doc(cache: &mut Cache, file_id: FileId) -> Result<(), Error> {
+    // Parse the file into the cache
```

Comment:
Nitpick: comments are a good thing, but some of those in this function are just very descriptive and don't add much to code understanding (parse the file in the cache: `cache.parse(file_id)?`, for every term: `for(..)`, etc.). It's a fine balance though, and a matter of taste, so please do as you see fit. Also, some do add value (like the one saying that the default settings for Comrak is essentially CommonMark).